src/components/layouts/Ide.js

```diff
@@ -6,46 +6,60 @@ import View from '../View';
 
 import * as layoutTypes from '../../constants/LayoutTypes.js';
 
-export const Ide = ({ editor, editorActions, user, authActions, scene, sceneActions, projectActions, courseActions, projects, courses, match, classroomActions, classrooms }) => (
-    <div className="App">
-        <Header
-            logging={authActions}
-            sceneActions={sceneActions}
-            actions={editorActions}
-            user={user}
-            scene={scene}
-            text={editor.text}
-            message={editor.message}
-            projectId={match.params.id}
-            match={match}
-            projectActions={projectActions}
-            courseActions={courseActions}
-            projects={projects}
-            courses={courses}
-            classroomActions={classroomActions}
-            classrooms={classrooms}
-            layoutType={layoutTypes.IDE}
-        />
-        <div className="row no-gutters">
-            {
-                scene.settings.viewOnly
-                    ?
-                    <div id="scene" className="col-12" >
-                        <View objects={editor.objects} sceneConfig={scene} assets={editor.assets} />
-                    </div>
-                    :
-                    <>
-                        <div id="interface" className="col-12 col-md-4" >
-                            <Editor text={editor.text} user={user} />
-                        </div>
-                        <div id="scene" className="col-12 col-md-8" >
-                            <View objects={editor.objects} sceneConfig={scene} assets={editor.assets} />
-                        </div>
-                    </>
-            }
-        </div>
-        <Footer />
-    </div>
-);
+class Ide extends React.Component {
+    render() {
+        return (
+            <div className="App">
+                <Header
+                    viewOnly = {this.props.scene.settings.view}
+                    logging={this.props.authActions}
+                    sceneActions={this.props.sceneActions}
+                    actions={this.props.editorActions}
+                    user={this.props.user}
+                    scene={this.props.scene}
+                    text={this.props.editor.text}
+                    message={this.props.editor.message}
+                    projectId={this.props.match.params.id}
+                    match={this.props.match}
+                    projectActions={this.props.projectActions}
+                    courseActions={this.props.courseActions}
+                    projects={this.props.projects}
+                    courses={this.props.courses}
+                    classroomActions={this.props.classroomActions}
+                    classrooms={this.props.classrooms}
+                    layoutType={layoutTypes.IDE}
+                />
+
+                <div className="row no-gutters">
+                    {
+                        /*
```

Comment:
Let's move comments about issues to Github. Code comments are best when they more about what the code does or quick issues. Feel free to make a reference to Github by citing the issue number.